File: P1/P1S1E2/lib/CarreraCarretera.rb
require('./ProductoCarrera.rb');

public
class CarreraCarretera < ProductoCarrera

	# @ReturnType String
	public
	def toString()
	  super + "\nTipo: Carretera\nPorcentaje de retirada: " + self.porcentajeRetirada.to_s + "%"
	end

	public
	def initialize()
		super(10)
	end
end

File: P1/P1S1E1/src/P1S1E1/CarreraCarretera.java
package P1S1E1;

public class CarreraCarretera extends ProductoCarrera {

        public CarreraCarretera(int nBicis, FactoriaAbstractaDeporte unaFact){
            super(nBicis, unaFact);
            super.setPorcentajeRetirada(10);
        }
        
        @Override
        public String toString(){
            return super.toString() + "\nTipo: Carretera\nPorcentaje de retirada: " + super.getPorcentajeRetirada() + "%";
        }
}